File: flipsum.go
package main

import (
	"fmt"
	"time"
)

const NUMBER = 1_000_000_000

func flipSum(start, end int) int {
	var flipSum int
	for n := end; n >= start; n-- {
		if n%2 == 0 {
			flipSum += n
		} else {
			flipSum -= n
		}
	}
	return flipSum
}

func main() {
	fmt.Printf("Flipsum %d\n", NUMBER)
	startTime := time.Now()
	result := flipSum(1, NUMBER)
	endTime := time.Now()

	fmt.Printf("Result = %d\n", result)
	fmt.Printf("Took %fs\n", endTime.Sub(startTime).Seconds())
}

File: flipsum.rs
use std::time::Instant;

const NUMBER: i32 = 1_000_000_000;

fn flip_sum(start: i32, end: i32) -> i32 {
    let mut flip_sum = 0;
    for n in (start..=end).rev() {
        if n % 2 == 0 {
            flip_sum += n;
        } else {
            flip_sum -= n;
        }
    }
    flip_sum
}

fn main() {
    println!("Flipsum {}", NUMBER);
    let start_time = Instant::now();
    let result = flip_sum(1, NUMBER);
    let end_time = Instant::now();

    println!("Result = {}", result);
    println!("Took {}s", (end_time - start_time).as_secs_f64());
}

